journal.py

```diff
@@ -86,49 +98,78 @@ def list_view(request):
     return {'entries': entries}
 
 
-@view_config(route_name='detail', renderer='templates/detail.jinja2')
+@view_config(route_name='detail',
+             xhr=True,
+             renderer='json')
+@view_config(route_name='detail',
+             xhr=False,
+             renderer='templates/detail.jinja2')
 def detail_view(request):
     entry = Entry.one(request.matchdict['id'])
-    html_text = markdown(entry.text, output_format='html5')
 
-    def my_highlight(matchobj):
-        return highlight(matchobj.group(0), PythonLexer(), HtmlFormatter())
+    if 'HTTP_X_REQUESTED_WITH' not in request.environ:
+        created = entry.created
+    else:
+        created = None
 
-    pattern = r'(?<=<code>)[\s\S]*(?=<\/code>)'
-    html_text = re.sub(pattern, my_highlight, html_text)
     return {
         'entry': {
             'id': entry.id,
             'title': entry.title,
-            'text': html_text,
-            'created': entry.created
+            'text': entry.mkdown,
+            'created': created
```

Comment:
talk to jonathan about the `__json__` method we discovered that Pyramid will use to encode arbitrary objects to JSON when you use the `"json"` renderer.  It's nifty!